api/src/controllers/post/mine.js

```diff
@@ -0,0 +1,68 @@
+const mongoose = require('mongoose');
+
+const postUtils = require('../../utils/post-utils');
+
+/**
+ * Create a new Mine record.
+ *
+ * @param {*} args
+ * @param {*} res
+ * @param {*} next
+ * @param {*} incomingObj Mine record to create
+ * @returns new Mine record
+ */
+exports.createRecord = async function (args, res, next, incomingObj) {
+  // TODO: Populate if needed.
+  const flavourFunctions = {};
+  return await postUtils.createRecordWithFlavours(args, res, next, incomingObj, this.createMaster, flavourFunctions);
+};
+
+/**
+ * Performs all operations necessary to create a master Mine record.
+ *
+ * @param {*} args
+ * @param {*} res
+ * @param {*} next
+ * @param {*} incomingObj Mine record to create
+ * @param {*} flavourIds array of flavour record _ids
+ * @returns created master Mine record
+ */
+exports.createMaster = function (args, res, next, incomingObj, flavourIds) {
+  let Mine = mongoose.model('MineBCMI');
+  let mine = new Mine();
+
+  mine._schemaName = 'MineBCMI';
+
+  // set permissions
+  mine.read = ['sysadmin'];
+  mine.write = ['sysadmin'];
+
+  // set data
+  incomingObj.name && (mine.name = incomingObj.name);
+  incomingObj.permitNumbers.length && (mine.permitNumbers = incomingObj.permitNumbers);
+  incomingObj.status && (mine.status = incomingObj.status);
+  incomingObj.commodities.length && (mine.commodities = incomingObj.commodities);
+  incomingObj.region && (mine.region = incomingObj.region);
+  incomingObj.location && (mine.location = incomingObj.location);
+  incomingObj.permittee && (mine.permittee = incomingObj.permittee);
+  incomingObj.type && (mine.type = incomingObj.type);
+  incomingObj.summary && (mine.summary = incomingObj.summary);
+  incomingObj.description && (mine.description = incomingObj.description);
+  incomingObj.links.length && (mine.links = incomingObj.links);
```

Comment:
We probably have to check that links is not null before calling .length, otherwise it could throw an error.
Same with the other .length calls.

```incomingObj.links && incomingObj.links.length ...```

Reply:
Updated